README.md

```diff
@@ -11,4 +11,73 @@ However, if exceptions occur frequently, consider adjusting guidelines (by creat
 
 ## Table of contents
 
-// TODO
+- [Git](#git)
+  - [Workflow](#workflow)
+  - [Commit messages](#commit-messages)
+
+## [Git](#git)
+
+### [Workflow](#workflow)
+
+1. Create a feature branch  
+
+    - Prefix branch name with `feature/` or `fix/` when applicable; otherwise, prefix with your initials, e.g. `tk/`
+    - Use kebabcase
+    - Include the issue number when applicable
+    - Briefly describe the change  
+
+    Examples: `feature/610-seelog-wrapper`, `fix/tequilapi-startup`, `tk/polish`
+
+2. Implement the changes (series of commits)
+    - It's best when commmits are clearly separated, i.e. each fix, change, refactoring, extra tests go into their own commits
+    - When fixing up a commit that already exists on your feature branch, try your best to squash the fix to the related commit so that we can avoid noise in git history
+      - If it's the last commit:
+
+        ```
+        $ git commit --amend
+        ```
+
+      - Otherwise, do [an interactive rebase](https://help.github.com/en/articles/about-git-rebase) to squash them into cohesive commits with good messages:
+
+        ```
+        $ git fetch origin
+        $ git rebase -i origin/master
+        ```
+3. Rebase frequently to incorporate upstream changes
```

Comment:
I'll point it here once again. 
Doing rebase for PR in GitHub does not keep history, and you have to review full PR from scratch, instead of only last commits. This is slowing down the PR review.

Reply:
It does keep history, if you click on _force-pushed_ it will show the diff from the previous branch tip. Example for this PR: https://github.com/mysteriumnetwork/devbook/compare/cbda7bd5af50ca8833e02c08d81a433274848280..764fbaa4b372e766ccd92e0e73d9bbd933df6eb0

Reply:
I think you are talking more about interactive rebase and editing existing commits. Rebase itself does nothing wrong.

Reply:
Yeah, I'm talking more about the review process, since commits were changed you don't know was it reviewed already or not.

Reply:
We'll need to consider options then - either remove the auto-stale'ing of reviews once it's been rebased(which seems bad) - or allow merging, but force squashing to master. If we want to move towards smaller PR's - it makes sense to squash to master.

Reply:
Squashing obfuscates history not less than tons of merge commits. Perhaps we can reconsider squashing _after_ we fully adopts small PRs?